### PR TITLE
B0, fw_info, secure_services: Small doc improvements

### PR DESCRIPTION
**Pull request by oyvindronningstad**

### CODE REVIEW

File: include/bl_validation.h [hunk 1]
```diff
@@ -18,6 +18,21 @@
 #include <stdbool.h>
 #include <fw_info.h>
 
+/**Function for validating firmware in place.
```
Comment (hakonfam) [inline include/bl_validation.h]: nit: Do we normally have a space here?  Ignore if no other change is needed.
```
/** Something
```

File: include/fw_info.h [hunk 2]
```diff
@@ -252,7 +252,14 @@ BUILD_ASSERT_MSG(ARRAY_SIZE(allowed_offsets) == FW_INFO_OFFSET_COUNT,
 	#error FW_INFO_OFFSET not set to one of the allowed values.
 #endif
 
-/* Search for the firmware_info structure inside the firmware. */
+
+/**Search for the firmware_info structure inside the firmware.
```
Comment (hakonfam) [inline include/fw_info.h]: nit as before

File: include/bl_validation.h (hunk 1 above)
Comment (oyvindronningstad) [reply include/bl_validation.h]: I think yours looks nicer and is more commonly used :). Thanks

File: include/bl_validation.h [hunk 3]
```diff
@@ -18,6 +18,21 @@
...
+ *
+ * @details This will run a series of checks on the fw_info contents, then
+ *          locate the validation info and check the signature of the image.
+ *
+ * @note If this function returns false, the fw_info struct can be invalidated
+ *       by modifying its 'valid' field, causing any subsequent call to this
+ *       function to quickly return false.
+ *
+ * @param[in] fw_address  Address the firmware is currently at.
+ * @param[in] fw_info     Firmware info structure belonging to the image.
```
Comment (hakonfam) [inline include/bl_validation.h]:
```suggestion
 * @param[in] fwinfo     Firmware info structure belonging to the image.
```

**Comment by hakonfam** [comment]
I saw that you requested review from b-gent, but do we really need doc-writers on this as its doxygen only?

File: include/bl_validation.h (hunk 3 above)
Comment (oyvindronningstad) [reply include/bl_validation.h]: Done

**Comment by oyvindronningstad** [comment]
@sigvartmh If you review and approve I think we can merge this.

**Pull request merged**